.chronus/changes/core-addParameterizedNextLink-2025-2-17-14-28-54.md

```diff
@@ -0,0 +1,7 @@
+---
+changeKind: feature
+packages:
+  - "@azure-tools/typespec-azure-core"
+---
+
+add internal scalar type `parameterizedNextLink` to handle next links that don't meet guidelines
```

Comment:
is the type supposed to be internal?

Reply:
would it be ok if we made it internal? you're right, right now the decorator is internal but the scalar is public. Can people use internal types in their specs?

Reply:
it warn if they use from the `Private` namespace. But we also have a `Legacy` namespace for bad patterns, maybe it belongs there

Reply:
yeah I think legacy makes sense, let me move it into there

Reply:
hmm not finding `Legacy`. I'm seeing `obsolete`, everything there's deprecated so not sure it's the one to use

Reply:
Legacy exists in Azure.ResourceManager, but not here.  I would suggest adding a Azure.Core.Legacy namespace specifically for this, I am sure it won't be the last thing we put in this namespace.

We should likely have a similar linting rule that emits a warning when components from this namespace are used.